src/components/new-request/search.vue

```diff
@@ -467,6 +499,19 @@ export default class Search extends Mixins(CommonMixin, NrAffiliationMixin, Sear
     return (this.isSocietyEnabled() && this.getEntityTypeCd === EntityTypes.SO)
   }
 
+  get isSocietyDisabled (): boolean {
+    this.openSocietyDialog()
+
+    let isSocietyDisabled = (
+      !this.isSocietyEnabled() &&
+      (this.getEntityTypeCd === EntityTypes.SO || this.getEntityTypeCd === EntityTypes.XSO)
+    )
+
+    this.setSocietiesModalVisible(isSocietyDisabled)
+
+    return isSocietyDisabled
+  }
+
```

Comment:
I'm quite confused by this getter (computed). It should not be doing multiple things (and esp not setting state when you "get" something).

I think you go back to my earlier suggestion of passing a prop to show/hide the dialog, and using an event handler when it's closed to clean up the variables.

After that, if you still need to call something when a variable changes then use a watcher method.

Reply:
Changed, thanks.